lib/java/com/google/android/material/divider/MaterialDivider.java

```diff
@@ -0,0 +1,193 @@
+
+
+  public MaterialDivider(Context context) {
+    this(context, null);
+  }
+
+  public MaterialDivider(Context context, AttributeSet attrs) {
+    this(context, attrs, R.attr.dividerStyle);
+  }
+
+  boolean isRtlEnabled;
+
+  AttributeSet initialAttrs;
+  int initialDefStyleAttr;
+
+  private static final int DEF_STYLE_RES = R.style.Widget_MaterialComponents_Divider;
+
+  public MaterialDivider(Context context, AttributeSet attrs, int defStyleAttr) {
+
+    super(wrap(context, attrs, defStyleAttr, DEF_STYLE_RES), attrs, defStyleAttr);
+    initialAttrs = attrs;
+    initialDefStyleAttr = defStyleAttr;
+    isRtlEnabled = ViewUtils.isLayoutRtl(this);
+
+    float alpha = attrs.getAttributeFloatValue("http://schemas.android.com/apk/res/android", "alpha", -1);
+    int color = attrs.getAttributeIntValue(android.R.attr.background, -1);
+
+    final TypedArray dividerStyleAttrs = getDividerStyleAttrs();
+    if (alpha == -1) {
+      alpha = dividerStyleAttrs.getFloat(ATTR_ALPHA, -1);
+    }
+    if (color == -1) {
+      color = dividerStyleAttrs.getColor(ATTR_COLOR, -1);
+    }
+
+    setAlpha(alpha);
+    setColor(color);
+  }
+
+  int[] ATTRS = {R.attr.dividerAlpha, R.attr.dividerColor, R.attr.dividerInset, R.attr.dividerSize};
+  private static final int ATTR_ALPHA = 0;
+  private static final int ATTR_COLOR = 1;
+  private static final int ATTR_INSET = 2;
+  private static final int ATTR_SIZE = 3;
+
+  private TypedArray getDividerStyleAttrs() {
+    TypedValue dividerStyle = new TypedValue();
+    getContext().getTheme().resolveAttribute(R.attr.dividerStyle, dividerStyle, true);
+    return getContext().obtainStyledAttributes(dividerStyle.resourceId, ATTRS);
+  }
+
+  //Runs when this view's layoutParams are available
+  @Override
+  protected void onAttachedToWindow() {
+
+    super.onAttachedToWindow();
+    initializeOrientation();
+    TypedArray dividerStyleAttrs = getDividerStyleAttrs();
+    TypedArray styledAttributes =
+        ThemeEnforcement.obtainStyledAttributes(getContext(), initialAttrs, R.styleable.MaterialDivider, initialDefStyleAttr, DEF_STYLE_RES);
+    LayoutParams layoutParams = getLayoutParams();
+
+    if (orientation != INVALID) {
+      int size = orientation == VERTICAL ? layoutParams.height : layoutParams.width;
+      if (size == WRAP_CONTENT) {
+        size = (int) dividerStyleAttrs.getDimension(ATTR_SIZE, -1);
+        setSize(size);
+      }
+    }
+
+    if (orientation == VERTICAL) {
+      int inset = (int) styledAttributes.getDimension(R.styleable.MaterialDivider_inset, -1);
+      if (inset == -1) {
+        inset = (int) dividerStyleAttrs.getDimension(ATTR_INSET, -1);
+      }
+      setInset(inset);
+    }
+
+    dividerStyleAttrs.recycle();
+    styledAttributes.recycle();
+  }
+
+  private static final int INVALID = -1;
+  private static final int HORIZONTAL = 0;
+  private static final int VERTICAL = 1;
+  private int orientation;
+
+  private void initializeOrientation() {
```

Comment:
Orientation should not be inferred from the layout params, why not add support for eg. a `dividerOrientation` attribute in the styleable so that it's explicit? Along with a getter/setter